src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -67,6 +67,113 @@ internal class FeatureManagementKeyValueAdapter : IKeyValueAdapter
                 keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}", false.ToString()));
             }
 
+            if (featureFlag.Variants != null)
+            {
+                for (int i = 0; i < featureFlag.Variants.Count; i++)
+                {
+                    FeatureVariant featureVariant = featureFlag.Variants[i];
+
+                    keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Variants}:{i}:Name", featureVariant.Name));
+
+                    if (featureVariant.ConfigurationValue != null)
+                    {
+                        keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Variants}:{i}:ConfigurationValue", featureVariant.ConfigurationValue));
+                    }
+
+                    if (featureVariant.ConfigurationReference != null)
+                    {
+                        keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Variants}:{i}:ConfigurationReference", featureVariant.ConfigurationReference));
+                    }
+
+                    if (featureVariant.StatusOverride != null)
+                    {
+                        keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Variants}:{i}:StatusOverride", featureVariant.StatusOverride));
+                    }
+                }
+            }
+
+            if (featureFlag.Allocation != null)
+            {
+                FeatureAllocation featureAllocation = featureFlag.Allocation;
+
+                if (featureAllocation.DefaultWhenDisabled != null)
+                {
+                    keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Allocation}:DefaultWhenDisabled", featureAllocation.DefaultWhenDisabled));
+                }
+
+                if (featureAllocation.DefaultWhenEnabled != null)
+                {
+                    keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Allocation}:DefaultWhenEnabled", featureAllocation.DefaultWhenEnabled));
+                }
+
+                if (featureAllocation.User != null)
+                {
+                    for (int i = 0; i < featureAllocation.User.Count; i++)
+                    {
+                        FeatureUserAllocation featureUserAllocation = featureAllocation.User[i];
```

Comment:
```suggestion
                        FeatureUserAllocation userAllocation = featureAllocation.User[i];
```

Reply:
Suggested because the current string is so similar to featureAllocation which is also defined.